api/routes/graphs.py

```
@@ -225,3 +227,51 @@ async def delete_graph(request: Request, graph_id: str):
+@graphs_router.get("/{graph_id}/user-rules", responses={401: UNAUTHORIZED_RESPONSE})
+@token_required
+async def get_graph_user_rules(request: Request, graph_id: str):
+    """Get user rules for the specified graph."""
+    try:
+        full_graph_id = _graph_name(request.state.user_id, graph_id)
+        user_rules = await get_user_rules(full_graph_id)
+        logging.info("Retrieved user rules length: %d", len(user_rules) if user_rules else 0)
+        return JSONResponse(content={"user_rules": user_rules})
+    except GraphNotFoundError:
+        return JSONResponse(content={"error": "Database not found"}, status_code=404)
+    except Exception as e:  # pylint: disable=broad-exception-caught
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)